src/Portable.Xaml/Portable.Xaml/XamlWriterInternalBase.cs

```diff
@@ -41,14 +41,31 @@ namespace Mono.Xaml
 namespace Portable.Xaml
 #endif
 {
-	abstract class XamlWriterInternalBase : IProvideValueTarget, IRootObjectProvider, IDestinationTypeProvider, IAmbientProvider
+	abstract class XamlWriterInternalBase : IProvideValueTarget, IRootObjectProvider, IDestinationTypeProvider
 	{
 		protected XamlWriterInternalBase(XamlSchemaContext schemaContext, XamlWriterStateManager manager)
+			: this(schemaContext, manager, null)
+		{
+		}
+
+		protected XamlWriterInternalBase(XamlSchemaContext schemaContext, XamlWriterStateManager manager, IAmbientProvider parentAmbientProvider = null)
 		{
+			this.parentAmbientProvider = parentAmbientProvider;
 			this.sctx = schemaContext;
 			this.manager = manager;
 			var p = new PrefixLookup(sctx) { IsCollectingNamespaces = true }; // it does not raise unknown namespace error.
-			service_provider = ValueSerializerContext.Create(p, schemaContext, this, this, this, this, this as IXamlObjectWriterFactory);
+			service_provider = ValueSerializerContext.Create(p, schemaContext, GetCurrentAmbientProvider, this, this, this, this as IXamlObjectWriterFactory);
+		}
+
+
+
+		protected virtual IAmbientProvider GetCurrentAmbientProvider()
+		{
+			IAmbientProvider basicAmbientProvider = new ObjectStatesAmbientProvider(object_states);
+			var ambientProvider =
+				parentAmbientProvider == null ? basicAmbientProvider
+					: new StackAmbientProvider(basicAmbientProvider, parentAmbientProvider);
+			return ambientProvider;
```

Comment:
This should be resolved when the ValueSerializerContext is created, and then just pass the instance of the provider to use.